src/gpu/intel/ocl/micro_sdpa.hpp

```diff
@@ -65,15 +65,23 @@ struct micro_sdpa_t : public gpu_primitive_t {
                         attn_mask_md()->dims[mask_k_index] == desc()->keys(),
                         VERBOSE_INVALID_BROADCAST, "attn_mask", mask_k_index);
             }
-            VDISPATCH_SDPA(utils::everyone_is(data_type::f16,
-                                   qry_md()->data_type, dst_md()->data_type),
+            VDISPATCH_SDPA(
+                    (utils::everyone_is(data_type::f16, qry_md()->data_type,
+                             dst_md()->data_type)
+                            || utils::everyone_is(data_type::bf16,
+                                    qry_md()->data_type, dst_md()->data_type)),
                     VERBOSE_UNSUPPORTED_DT);
             VDISPATCH_SDPA(
                     utils::one_of(key_md()->data_type, f16, u8, s8, u4, s4),
                     VERBOSE_UNSUPPORTED_DT);
             VDISPATCH_SDPA(
                     utils::one_of(val_md()->data_type, f16, u8, s8, u4, s4),
                     VERBOSE_UNSUPPORTED_DT);
+            VDISPATCH_SDPA(
+                    IMPLICATION(qry_md()->data_type == data_type::bf16,
+                            utils::everyone_is(data_type::bf16,
+                                    val_md()->data_type, key_md()->data_type)),
+                    VERBOSE_UNSUPPORTED_DT);
```

Comment:
```suggestion
            VDISPATCH_SDPA(
                    IMPLICATION(qry_md()->data_type == data_type::bf16,
                            utils::everyone_is(data_type::bf16,
                                    val_md()->data_type, key_md()->data_type, attn_mask_md()->data_type, scale_dt)),
                    "key(%s) and value(%s) tensors must be bf16 if query is bf16.", key_md()->data_type, val_md()->data_type); // Some assembly required
```

You should add atn_mask_md and scale_dt to this check because you use CONVERT_FLOAT_T for those values.

Also, it would be nice to have a message useful message

Reply:
I might have to add another VERBOSE tag for that.